File: dal-dao-gen/dal_ftp/63/cs/Dao/Test2Dao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using Arch.Data;
using Arch.Data.DbEngine;
using com.ctrip.dal.test.test4.Entity.DataModel;
using com.ctrip.dal.test.test4.Interface.IDao;

namespace com.ctrip.dal.test.test4.Dao
{
   /// <summary>
    /// 更多DAL接口功能，请参阅DAL Confluence，地址：
    /// http://conf.ctripcorp.com/display/SysDev/Dal+Fx+API
    /// </summary>
    public partial class Test2Dao : ITest2Dao
    {
        readonly BaseDao baseDao = BaseDaoFactory.CreateBaseDao("AccCorpDB_INSERT_1");
        

        /// <summary>
        /// 获取所有Test2信息
        /// </summary>
        /// <returns>Test2列表</returns>
        public IList<Test2> GetAll()
        {
            try
            {
                return baseDao.GetAll<Test2>();
            }
            catch (Exception ex)
            {
                throw new DalException("调用Test2Dao时，访问GetAll时出错", ex);
            }
        }
        
        /// <summary>
        /// 取得总记录数
        /// </summary>
        /// <returns>记录数</returns>
        public long Count()
        {
            try
            {
                String sql = "SELECT count(1) from test2  with (nolock)  ";

                object obj = baseDao.ExecScalar(sql);
                long ret = Convert.ToInt64(obj);
                return ret;
            }
            catch (Exception ex)
            {
                throw new DalException("调用Test2Dao时，访问Count时出错", ex);
            }
        }
        
        /// <summary>
        ///  检索Test2，带翻页
        /// </summary>
        /// <param name="obj">Test2实体对象检索条件</param>
        /// <param name="pagesize">每页记录数</param>
        /// <param name="pageNo">页码</param>
        /// <returns>检索结果</returns>
        public IList<Test2> GetListByPage(Test2 obj, int pagesize, int pageNo)
        {
             try
            {
                var dic = new StatementParameterCollection();
                StringBuilder sbSql = new StringBuilder(200);

                
                 //计算ROWNUM
                int fromRownum = (pageNo - 1) * pagesize + 1;
                int endRownum = pagesize * pageNo;
                 sbSql.Append("WITH CTE AS ("); //WITH CTE 开始
                sbSql.Append("select row_number() over(order by AccBalanceID desc ) as rownum, ");
                sbSql.Append(@"AccBalanceID, BatchNo, AccountID, CreateTime, StartDate, EndDate, recReturn, ReportCompletionDay, ContractFirmDay, ContractDay, ReConfirmDate, ContractDate, LastAuditDate, SendRptDate, AuditDate, ReceiveDate, RptDate, ConfirmDate, BalDate, InvoiceDate, ConfirmFlag, LastAuditFlag, SettleFlag from test2 (nolock) ");

                //包含查询条件
                //StringBuilder whereCondition = new StringBuilder();
                //if (!string.IsNullOrEmpty(obj.Name))
                //{
                //    //人名
                //    whereCondition.Append("Where Name like @Name ");
                //    dic.AddInParameter("@Name", DbType.String, "%" + obj.Name + "%");
                //}
                //sbSql.Append(whereCondition);

                sbSql.Append(")"); //WITH CTE 结束

                // 用 CTE 完成分页
                sbSql.Append(@"select AccBalanceID, BatchNo, AccountID, CreateTime, StartDate, EndDate, recReturn, ReportCompletionDay, ContractFirmDay, ContractDay, ReConfirmDate, ContractDate, LastAuditDate, SendRptDate, AuditDate, ReceiveDate, RptDate, ConfirmDate, BalDate, InvoiceDate, ConfirmFlag, LastAuditFlag, SettleFlag from CTE Where rownum between @from and @end");
                dic.AddInParameter("@from", DbType.Int32, fromRownum);
                dic.AddInParameter("@end", DbType.Int32, endRownum);
                IList<Test2> list = baseDao.SelectList<Test2>(sbSql.ToString(), dic);

                return list;
            }
            catch (Exception ex)
            {
                throw new DalException("调用Test2Dao时，访问GetListByPage时出错", ex);
            }
        }






        
    }
}

File: dal-dao-gen/dal_ftp/63/java/Dao/Test2Dao.java
package com.ctrip.dal.test.test4;


import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Timestamp;
import java.util.List;

import com.ctrip.platform.dal.dao.DalClient;
import com.ctrip.platform.dal.dao.DalClientFactory;
import com.ctrip.platform.dal.dao.DalHints;
import com.ctrip.platform.dal.dao.DalRowMapper;
import com.ctrip.platform.dal.dao.StatementParameters;
import com.ctrip.platform.dal.dao.helper.DalRowMapperExtractor;
import com.ctrip.platform.dal.dao.helper.DalScalarExtractor;

public class Test2Dao {

	private static final String DATA_BASE = "AccCorpDB_INSERT_1";
	private static final String ALL_SQL_PATTERN = "SELECT * FROM test2";
	private static final String COUNT_SQL_PATTERN = "SELECT count(1) from test2";
	private static final String PAGE_MYSQL_PATTERN = "SELECT * FROM test2 WHERE LIMIT %s, %s";
	private static final String PAGE_SQL_PATTERN = "WITH CTE AS (select *, row_number() over(order by AccBalanceID desc ) as rownum" 
			+" from test2 (nolock)) select * from CTE where rownum between %s and %s";
			
	private DalClient client;
	private Test2RowMapper mapper;
	private DalRowMapperExtractor<Test2> extractor;
	private DalScalarExtractor scalarExtractor;
	
	/**
	 * Initialize the instance of Hotel2GenDao
	 */
	public Test2Dao()
	{
		this.client = DalClientFactory.getClient(DATA_BASE);
		this.mapper = new Test2RowMapper();
		this.extractor = new DalRowMapperExtractor<Test2>(this.mapper);
		this.scalarExtractor = new DalScalarExtractor();
	}

	/**
	  *Get all Test2 instances
	  *@return 
	  *     Test2 collection
	**/
	public List<Test2> getAll() throws SQLException
	{
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		List<Test2> result = null;
		result = this.client.query(ALL_SQL_PATTERN, parameters, hints, extractor);
		return result;
	}
	
	/**
	  *Get the count of Test2 instances
	  *@return 
	  *     the Test2 records count
	**/
	public long Count() throws SQLException
	{
		StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		long result = (Long)this.client.query(COUNT_SQL_PATTERN, parameters, hints, scalarExtractor);
		return result;
	}
	
	public List<Test2> getListByPage(Test2 obj, int pagesize, int pageNo) throws SQLException
	{
		if(pageNo < 1 || pagesize < 1) 
			throw new SQLException("Illigal pagesize or pageNo, pls check");
		
        StatementParameters parameters = new StatementParameters();
		DalHints hints = new DalHints();
		
		String sql = "";
		int fromRownum = (pageNo - 1) * pagesize + 1;
        int endRownum = pagesize * pageNo;
		sql = String.format(PAGE_SQL_PATTERN, fromRownum, endRownum);
		return this.client.query(sql, parameters, hints, extractor);
	}

	/**
	  * Map the sql result-set to Test2 instance
	**/
	private class Test2RowMapper implements DalRowMapper<Test2> {

		@Override
		public Test2 map(ResultSet rs, int rowNum) throws SQLException {
			Test2 pojo = new Test2();
			
			pojo.setAccBalanceID((Integer)rs.getObject("AccBalanceID"));
			pojo.setBatchNo((String)rs.getObject("BatchNo"));
			pojo.setAccountID((Integer)rs.getObject("AccountID"));
			pojo.setCreateTime((Timestamp)rs.getObject("CreateTime"));
			pojo.setStartDate((String)rs.getObject("StartDate"));
			pojo.setEndDate((String)rs.getObject("EndDate"));
			pojo.setRecReturn((BigDecimal)rs.getObject("recReturn"));
			pojo.setReportCompletionDay((Timestamp)rs.getObject("ReportCompletionDay"));
			pojo.setContractFirmDay((Timestamp)rs.getObject("ContractFirmDay"));
			pojo.setContractDay((Timestamp)rs.getObject("ContractDay"));
			pojo.setReConfirmDate((Timestamp)rs.getObject("ReConfirmDate"));
			pojo.setContractDate((Timestamp)rs.getObject("ContractDate"));
			pojo.setLastAuditDate((Timestamp)rs.getObject("LastAuditDate"));
			pojo.setSendRptDate((Timestamp)rs.getObject("SendRptDate"));
			pojo.setAuditDate((Timestamp)rs.getObject("AuditDate"));
			pojo.setReceiveDate((Timestamp)rs.getObject("ReceiveDate"));
			pojo.setRptDate((Timestamp)rs.getObject("RptDate"));
			pojo.setConfirmDate((Timestamp)rs.getObject("ConfirmDate"));
			pojo.setBalDate((Timestamp)rs.getObject("BalDate"));
			pojo.setInvoiceDate((Timestamp)rs.getObject("InvoiceDate"));
			pojo.setConfirmFlag((String)rs.getObject("ConfirmFlag"));
			pojo.setLastAuditFlag((String)rs.getObject("LastAuditFlag"));
			pojo.setSettleFlag((String)rs.getObject("SettleFlag"));

			return pojo;
		}
	}
}
